src/syntax/syntax.rkt

```diff
@@ -3,21 +3,62 @@
 (require math/flonum math/base math/bigfloat math/special-functions rival)
 (require "../common.rkt" "../interface.rkt" "../errors.rkt" "../float32.rkt" "types.rkt")
 
-(provide constant? variable? operator? operator-info constant-info get-operator-itype
+(provide constant? variable? operator? operator-info constant-info get-operator-arity
          get-parametric-operator parametric-operators parametric-operators-reverse
          get-parametric-constant parametric-constants parametric-constants-reverse
          *unknown-ops* *loaded-ops*
          repr-conv? rewrite-repr-op? get-repr-conv)
 
 (module+ internals 
-  (provide operators constants define-constant define-operator infix-joiner
-           register-constant! register-operator!))
+  (provide operator-impls constant-impls infix-joiner
+           define-constant-impl define-operator-impl
+           register-constant-impl! register-operator-impl!
+           define-constant define-operator
+           register-impl! register-operator!))
+
 
 (module+ test (require rackunit))
 
-;; Constants' values are functions because they may depend on (bf-precision)
+;; Abstract constant table
+;; Implementations inherit attributes
```

Comment:
Wonder if life would be easier if constants were internally represented as nullary operators.